assignments/lambda-classes.js

```diff
@@ -1 +1,193 @@
+* Person receives `speak` as a method.
+* This method logs out a phrase `Hello my name is Fred, I am from Bedrock` where `name` and `location` are the object's own props
+
+#### Instructor
+
+* Now that we have a Person as our base class, we'll build our Instructor class.
+* Instructor uses the same attributes that have been set up by Person
+* Instructor has the following unique props:
+  * `specialty` what the Instructor is good at i.e. 'redux'
+  * `favLanguage` i.e. 'JavaScript, Python, Elm etc.'
+  * `catchPhrase` i.e. `Don't forget the homies`
+* Instructor has the following methods:
+  * `demo` receives a `subject` string as an argument and logs out the phrase 'Today we are learning about {subject}' where subject is the param passed in.
+  * `grade` receives a `student` object and a `subject` string as arguments and logs out '{student.name} receives a perfect score on {subject}'
+
+#### Student
+
+* Now we need some students!
+* Student uses the same attributes that have been set up by Person
+* Student has the following unique props:
+  * `previousBackground` i.e. what the Student used to do before Lambda School
+  * `className` i.e. CS132
+  * `favSubjects`. i.e. an array of the student's favorite subjects ['Html', 'CSS', 'JavaScript']
+* Student has the following methods:
+  * `listsSubjects` a method that logs out all of the student's favoriteSubjects one by one.
+  * `PRAssignment` a method that receives a subject as an argument and logs out that the `student.name has submitted a PR for {subject}`
+  * `sprintChallenge` similar to PRAssignment but logs out `student.name has begun sprint challenge on {subject}`
+
+#### Project Manager
+
+* Now that we have instructors and students, we'd be nowhere without our PM's
+* ProjectManagers are extensions of Instructors
+* ProjectManagers have the following unique props:
+  * `gradClassName`: i.e. CS1
+  * `favInstructor`: i.e. Sean
+* ProjectManagers have the following Methods:
+  * `standUp` a method that takes in a slack channel and logs `{name} announces to {channel}, @channel standy times!​​​​​
+  * `debugsCode` a method that takes in a student object and a subject and logs out `{name} debugs {student.name}'s code on {subject}`
+
+#### Stretch Problem
+*/
+
+class Person {
+    
+  constructor(attributes) {
+  this.name = attributes.name;
+  this.age = attributes.age;
+  this.location = attributes.location;
+  this.gender = attributes.gender;
+  }
+
+  speak() {
+    console.log(` Hello, my name is ${this.name}, I am from ${this.location}.`)
+  }
+  
+}
+
+class Instructor extends Person {
+  constructor(attributes) {
+    super(attributes);
+    this.specialty = attributes.specialty;
+    this.favLanguage = attributes.favLanguage;
+    this.catchPhrase = attributes.catchPhrase;
+  }
+
+  demo(subject) {
+    console.log(`Today we are learning about ${subject}`)
+  }
+
+  grade(subject) {
+    console.log(`${Student.name} has recieved a perfect score on ${subject}`)
```

Comment:
This is very close. But you don't want to refer to the Capital "S" Student class constructor. You want to just use small "s" student as a variable that you can pass any student's name in to. 

```suggestion
    console.log(`${student.name} has recieved a perfect score on ${subject}`)
```

Reply:
If you wanted to call this down below, you could use:

fred.grade(welma, "Ruby");